sirepo/server.py

```diff
@@ -732,75 +738,35 @@ def javascript_redirect(redirect_uri):
     )
 
 
-def session_user(*args, **kwargs):
-    """Get/set the user from the Flask session
+def set_session_user(user):
+    """Set the user for the Flask session
+    """
+    assert user
+    flask.session[_SESSION_KEY_USER] = user
+    _wsgi_app.set_log_user(user)
+
 
-    With no positional arguments, is a getter. Else a setter.
+def session_user(checked=True):
+    """Get the user from the Flask session
 
     Args:
-        user (str): if args[0], will set the user; else gets
         checked (bool): if kwargs['checked'], assert the user is truthy
-        environ (dict): session environment to use instead of `flask.session`
 
     Returns:
         str: user id
     """
-    environ = kwargs.get('environ', None)
-    session = environ.get(_ENVIRON_KEY_BEAKER) if environ else flask.session
-    if args:
-        session[_SESSION_KEY_USER] = args[0]
-        _wsgi_app.set_log_user(args[0])
-    res = session.get(_SESSION_KEY_USER)
-    if not res and kwargs.get('checked', True):
-        raise KeyError(_SESSION_KEY_USER)
+    res = flask.session.get(_SESSION_KEY_USER)
```

Comment:
What about the case where we want to create a user. It needs to assert that the cookie sentinel is there.

Reply:
The existing code in simulation_db catches the KeyError and creates a user. That is why server.session_user() only raises KeyError if the cookie is valid. Otherwise it raises forbidden.